### PR TITLE
Add option for requiring a sudo password in U17.04

### PR DESCRIPTION
**Pull request by ttheune**
Make minor PEP8 updates

### CODE REVIEW

**Comment by ttheune** [comment]
First pass at adding the option, just done in Ubuntu 17.04; Please provide feedback

File: linux/ubuntu/17.04/foxpass_setup.py [hunk 1]
```diff
@@ -193,46 +195,55 @@ def augment_sshd_config():
 
 def augment_pam():
     if not file_contains('/etc/pam.d/common-session', r'pam_mkhomedir\.so'):
-         with open('/etc/pam.d/common-session', "a") as w:
-             w.write('session required                        pam_mkhomedir.so umask=0022 skel=/etc/skel\n')
+            with open('/etc/pam.d/common-session', "a") as w:
```
[the remark this answers is not in the document]
Comment (ttheune) [reply linux/ubuntu/17.04/foxpass_setup.py]: Odd, the pep8 linter says that this is the correct indent.

File: linux/ubuntu/17.04/foxpass_setup.py [hunk 2]
```diff
@@ -193,46 +195,55 @@ def augment_sshd_config():
...
 
 def fix_nsswitch():
     os.system("sed -i 's/passwd:.*/passwd:         compat ldap/' /etc/nsswitch.conf")
     os.system("sed -i 's/group:.*/group:          compat ldap/' /etc/nsswitch.conf")
     os.system("sed -i 's/shadow:.*/shadow:         compat ldap/' /etc/nsswitch.conf")
 
+
 # give "sudo" and chosen sudoers groups sudo permissions without password
-def fix_sudo(sudoers):
-    os.system("sed -i 's/^%sudo\tALL=(ALL:ALL) ALL/%sudo ALL=(ALL:ALL) NOPASSWD:ALL/' /etc/sudoers")
```
[the remark this answers is not in the document]
Comment (ttheune) [reply linux/ubuntu/17.04/foxpass_setup.py]: Just there for backward compatibility.  I'm happy to leave it alone.

File: linux/ubuntu/17.04/foxpass_setup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aren) [reply linux/ubuntu/17.04/foxpass_setup.py]: It was at 5. You made it 8, but it should be 4.

[the remark this answers is not in the document]
Comment (ttheune) [reply linux/ubuntu/17.04/foxpass_setup.py]: Ugh, my linter *definitely* should have picked that up.

File: linux/ubuntu/17.04/foxpass_setup.py [hunk 3]
```diff
@@ -193,46 +199,51 @@ def augment_sshd_config():
...
             w.write('\n#includedir /etc/sudoers.d\n')
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
+            w.write('# Adding Foxpass group to sudoers\n\
+            %{sudo} ALL=(ALL:ALL){nopasswd}'.format(sudo=sudoers,
+                                                    nopasswd='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
```
Comment (aren) [inline linux/ubuntu/17.04/foxpass_setup.py]: change `{nopasswd}` to `{commands}`

File: linux/ubuntu/17.04/foxpass_setup.py [hunk 4]
```diff
@@ -27,11 +27,13 @@
 import argparse
 from datetime import datetime
 import os
-import os.path
 import re
 import sys
 import urllib3
 
+from datetime import datetime
```
Comment (aren) [inline linux/ubuntu/17.04/foxpass_setup.py]: this is already included 6 lines up.

Comment (ttheune) [reply linux/ubuntu/17.04/foxpass_setup.py]: Put that there in the PEP8 cleanup and missed removing the previous call, have done so.  GitHub isn't picking up that change.

Comment (aren) [reply linux/ubuntu/17.04/foxpass_setup.py]: huh, still hasn't come through.

Comment (ttheune) [reply linux/ubuntu/17.04/foxpass_setup.py]: github says I pushed it 6 hours ago (e0791c8)

Comment (aren) [reply linux/ubuntu/17.04/foxpass_setup.py]: oh i see. i was expecting you to remove the one above this comment string, not the one on top. datetime is a system lib, so it belongs in the top group.

Comment (ttheune) [reply linux/ubuntu/17.04/foxpass_setup.py]: Ahh, for some reason I thought that sub-class imports went in their own section.

File: linux/ubuntu/17.04/foxpass_setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ttheune) [reply linux/ubuntu/17.04/foxpass_setup.py]: Did you want this changed?

[the remark this answers is not in the document]
Comment (aren) [reply linux/ubuntu/17.04/foxpass_setup.py]: no, backwards compatibility is good.

**Pull request merged**